### PR TITLE
Before performing an RPC service lookup, normalize the URL path

### LINKED ISSUE #650: Double slash (``/'') should be equivalent to single slash in RPCs
For example:
```
:; curl -s http://127.0.0.1:8732/chains/main/blocks//head/header
Compat { error: ErrorMessage { msg: "Invalid block_id_param: , reason: missing checksum" } }
:;  curl -s http://127.0.0.1:8732/chains/main/blocks/head/header
{"hash":"BLockGenesisGenesisGenesisGenesisGenesisf79b5d1CoW2","chain_id":"NetXdQprcVkpaWU","level":0,"proto":0,"predecessor":"BLockGenesisGenesisGenesisGenesisGenesisf79b5d1CoW2","timestamp":"2018-06-30T16:07:32Z","validation_pass":0,"operations_hash":"LLoZS2LW3rEi7KYU4ouBQtorua37aWWCtpDmv1n2x3xoKi6sVXLWp","fitness":[],"context":"CoVNUgnfVocNYM2HNkHxMMcZdzqFDkCQ5vEGa3gsdpkidoA1QPZo","protocol":"PrihK96nBAFSxVL1GLJTVhu9YnzkMFiBeuJRPA8NwuZVZCE1L6i"}
```
The former command should likely return the same as the latter command. This would match the behaviour of Nomadic Labs `tezos-node` which would help make sure that scripts that work against NL `tezos-node` would work without update with `tezedge`.

### PR DESCRIPTION
**Pull request by tizoc**
Fixes #650

### CODE REVIEW

File: rpc/src/server/mod.rs [hunk 1]
```diff
@@ -210,6 +214,19 @@ pub fn spawn_server(
         }))
 }
 
+/// Normalizes the request path
+fn normalize_path(original_path: &str) -> Option<String> {
```
Comment (bkontur) [inline rpc/src/server/mod.rs:218]: @tizoc 
Could you, please, add here some easy unit-test with some corner cases?
Because from description "Normalizes the request path" it is not very obvious, what it is doing :)

**Comment by bkontur** [comment]
@tizoc 
I just forced push rpc-cache-tweaks here, so closing this

File: rpc/src/server/mod.rs (hunk 1 above)
Comment (tizoc) [reply rpc/src/server/mod.rs:218]: @bkontur yes, but first I have to understand exactly what hyper and `PathTree` do (and don't do) with the path to understand what needs to be done here (and what doesn't). The current fix is a quick hack to fix multiple `/`s in the path. It is also important that the normalization behaves the same as what Octez does.

Comment (tizoc) [reply rpc/src/server/mod.rs:218]: Or to put it another way, the actual solution is probably much more complicated than this and requires some research, this one here is not the final solution.
